Core/HLE/scePsmf.cpp

```diff
@@ -232,9 +232,14 @@ class PsmfPlayer {
 	}
 
 	bool HasReachedEnd() {
-		bool videoPtsEnd = (s64)psmfPlayerAvcAu.pts >= (s64)totalDurationTimestamp - VIDEO_FRAME_DURATION_TS;
-		// If we're out of video data and have no audio, it's over even if the pts isn't there yet.
-		return videoPtsEnd || (mediaengine->IsVideoEnd() && mediaengine->IsNoAudioData());
+		if (psmfPlayerLibVersion >= 0x05050010) {
```

Comment:
Did some testing - this looks incorrect.  I added it in #6125, but I can't reproduce it now.  I think that part may not have been carefully based on tests.

From my current testing, no version of firmware looks at the "last timestamp" value in the file.  It always just waits until the end of the file.

I'm going to merge this, since it's basically right, but remove the if in a second commit.

-[Unknown]